File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        const int ReverseCommand = 16;

        Func<int,bool> isCode = (commandToEvaluate) => (commandValue & commandToEvaluate) > 0;

        Dictionary<int, string> handshake = new Dictionary<int, string>
        {
            { 1, "wink" },
            { 2, "double blink" },
            { 4, "close your eyes" },
            { 8, "jump"}
        };

        var commands = handshake.Keys.Where(code => isCode(code))
                                     .Select(code => handshake[code]);

        return isCode(ReverseCommand) ? commands.Reverse().ToArray() : commands.ToArray();
    }
}

File: go/secret-handshake/secret_handshake.go
package secret

func Handshake(code uint) []string {
	secretCodes := []string{}
	actionCodes := []string{"wink", "double blink", "close your eyes", "jump"}

	var result uint
	var maskedCheck uint

	for codeIndex, item := range actionCodes {
		maskedCheck = 1 << uint(codeIndex)
		result = maskedCheck & code

		if result == maskedCheck {
			secretCodes = append(secretCodes, item)
		}
	}

	maskedCheck = 1 << uint(4)
	result = maskedCheck & code

	if result == maskedCheck {
		for i, j := 0, len(secretCodes)-1; i < j; i, j = i+1, j-1 {
			secretCodes[i], secretCodes[j] = secretCodes[j], secretCodes[i]
		}
	}

	return secretCodes
}
